src/EnergyPlus/UnitarySystem.cc

```diff
@@ -0,0 +1,4299 @@
+                index = loop;
+                found = true;
+                break;
+            }
+        }
+        if (!found) {
+            ShowSevereError("getDesignSpecMSHPIndex: did not find UnitarySystemPerformance:Multispeed name =" + objectName + ". Check inputs");
+        }
+        return index;
+    }
+
+    int getUnitarySystemIndex(       // lookup vector index for UnitarySystem object name in object array EnergyPlus::UnitarySystems::unitarySys
+        std::string const &objectName // IDF name in input
+    )
+    {
+        int index = -1;
+        bool found = false;
+        for (std::size_t loop = 0; loop < unitarySys.size(); ++loop) {
+            UnitarySys *thisUnitarySysObjec = &unitarySys[loop];
+            if (UtilityRoutines::SameString(objectName, thisUnitarySysObjec->name)) {
+                index = loop;
+                found = true;
+                break;
+            }
+        }
+        return index;
+    }
+
+    void UnitarySys::initUnitarySystems(bool const firstHVACIteration,
+                                        int const &AirLoopNum,
+                                        bool const &FirstHVACIteration,
+                                        Optional_int_const OAUnitNum,
+                                        Optional<Real64 const> OAUCoilOutTemp)
+    {
+
+        if (myOneTimeFlag) {
+            // initialize or allocate something once
+            myOneTimeFlag = false;
+        }
+    }
+
+    void UnitarySys::getUnitarySystemInput(std::string const &objectName)
+    {
+
+        bool errorsFound(false);
+
+        UnitarySys::getUnitarySystemInputData(objectName, errorsFound);
+
+        if (errorsFound) {
+            ShowSevereError("getUnitarySystemInputData: did not find UnitarySystem. Check inputs");
+        }
+    }
+
+    void UnitarySys::getUnitarySystemInputData(std::string const &objectName, bool errorsFound)
+    {
+
+        static std::string const getUnitarySystemInput("getUnitarySystemInputData");
+        static std::string const unitarySysHeatPumpPerformanceObjectType("UnitarySystemPerformance:Multispeed");
+
+        std::string cCurrentModuleObject = "UnitarySystem";
+
+        auto const instances = inputProcessor->epJSON.find(cCurrentModuleObject);
+        if (instances == inputProcessor->epJSON.end()) {
+            ShowSevereError("getUnitarySystemInputData: did not find UnitarySystem object in input file. Check inputs");
+            errorsFound = true;
+        } else {
+            auto &instancesValue = instances.value();
+            for (auto instance = instancesValue.begin(); instance != instancesValue.end(); ++instance) {
+                
+                auto const &thisObjectName = instance.key();
+                // only get the current data
+                if (!UtilityRoutines::SameString(objectName, thisObjectName) || getUnitarySystemIndex(objectName) > -1) continue;
```

Comment:
this allows multiple calls, if not thisOjbectName or it's already gotten then cycle. So get only 1 data set. Still need to resolve UnitarySystems::getInputOnceFlag and will probably make it a local to UnitarySys.